h/admin/views/users.py

```diff
@@ -69,23 +63,56 @@ def users_activate(request):
                                     _query=(('username', user.username),)))
 
 
+@view_config(route_name='admin_users_rename',
+             request_method='POST',
+             permission='admin_users')
+def users_rename(request):
+    user = _form_request_user(request)
+
+    if user is None:
+        return httpexceptions.HTTPFound(
+            location=request.route_path('admin_users'))
+
+    old_username = user.username
+    new_username = request.params.get('new_username')
+
+    try:
+        svc = request.find_service(name='rename_user')
+        svc.check(user.id, new_username)
+
+        worker.rename_user.delay(user.id, new_username)
+
+        request.session.flash(
+            'The user "%s" will be renamed to "%s" in the backgroud. Refresh this page to see if it\'s already done' %
+            (old_username, new_username), 'success')
+
+        return httpexceptions.HTTPFound(
+            location=request.route_path('admin_users',
+                                        _query=(('username', new_username),)))
+
+    except (UserRenameError, ValueError) as e:
+        request.session.flash(str(e), 'error')
+        return httpexceptions.HTTPFound(
+            location=request.route_path('admin_users',
+                                        _query=(('username', old_username),)))
+
+
 @view_config(route_name='admin_users_delete',
              request_method='POST',
              permission='admin_users')
 def users_delete(request):
-    username = request.params.get('username')
-    user = models.User.get_by_username(request.db, username)
+    user = _form_request_user(request)
 
     if user is None:
+        return httpexceptions.HTTPFound(
+            location=request.route_path('admin_users'))
+
+    try:
+        delete_user(request, user)
         request.session.flash(
-            'Cannot find user with username %s' % username, 'error')
```

Comment:
Oh, I see, it's moved into the `_form_request_user` function.

Perhaps these view functions would be cleaner if `_form_request_user` threw a named exception if the user wasn't found, and then you added an exception view to handle that scenario. See `h/accounts/views.py` for examples of exception views.